src/compiler/factory/nodeFactory.ts

```diff
@@ -2843,6 +2843,8 @@ namespace ts {
         // @api
         function createTemplateLiteralLikeNode(kind: TemplateLiteralToken["kind"], text: string, rawText: string | undefined, templateFlags: TokenFlags | undefined) {
             const node = createBaseToken<TemplateLiteralLikeNode>(kind);
+            rawText ??= JSON.stringify(text).slice(1,-1).replace(/\\.|\$/g, s =>

```

Comment:
I don't think we should try to synthesize `rawText` here. Generally when `rawText` is `undefined` we intend to pull the "raw" version of the template literal from the source file during emit. `rawText` was added to allow API consumers to pass their own "raw" text, and in those cases it goes through `createTemplateLiteralLikeNodeChecked` which verifies that the "cooked" text matches the supplied "raw" text.

If someone were to pass the string `"\t	"` (byte sequence `5c 74 09`) for `text`, this would generate a "raw" text of `"\t\t"`. There's no way to be 100% sure what the user intended, so its better to be explicit.

I think the issue is actually in the implementation of the refactor, which should be reading the raw source text of the string literal rather than `node.text`.

Reply:
Its also important that `rawText` remains `undefined` for our tagged template transform to accurately get the actual raw text from the source file for non-user-created nodes:

https://github.com/microsoft/TypeScript/blob/eb3645f16b66ed05110ea957c77773cea739bae8/src/compiler/transformers/taggedTemplate.ts#L78-L84